odoo_tools/utils/git.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import os
+import subprocess
+
+from . import ui
+from .path import build_path
+
+
+def get_odoo_core(hash, dest="src/odoo", org="odoo", branch=None):
+    dest = build_path(dest)
+    _clone_or_fetch_repo(org, "odoo", branch, dest)
+    ui.echo(f"Checking out {hash}")
+    subprocess.run(["git", "-C", dest, "checkout", hash], check=True)
```

Comment:
what if we have git-autoshare-clone? Maybe worth adding dynamic support for it? Or just a TODO + roadmap

Reply:
yes